modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java

```diff
@@ -195,71 +202,116 @@ public static String toSqlType(ColumnType columnType) {
             case INT64:
-                return Long.MAX_VALUE - base;
+                return RND.nextLong(Integer.MAX_VALUE + 1, Long.MAX_VALUE) + 1;
             case FLOAT:
-                return Float.MAX_VALUE - (short) base + ((float) base / 1000);
+                // copy-paste from JDK 21 jdk.internal.util.random.RandomSupport.boundedNextFloat(java.util.random.RandomGenerator, float)
+                float bound = Float.MAX_VALUE;
+                float r = RND.nextFloat();
+                r = r * bound;
+                if (r >= bound) {
+                    r = Math.nextDown(bound);
+                }
+                return r;
             case DOUBLE:
-                return Double.MAX_VALUE - base + ((double) base / 1000);
+                return RND.nextDouble(Float.MAX_VALUE + Double.MIN_NORMAL, Double.MAX_VALUE);
             case STRING:
-                return "str_" + base;
+                return IgniteTestUtils.randomString(RND, precision);
             case BYTE_ARRAY:
-                return new byte[]{(byte) base, (byte) (base + 1), (byte) (base + 2)};
+                return IgniteTestUtils.randomBytes(RND, precision);
             case NULL:
                 return null;
             case DECIMAL:
-                return BigDecimal.valueOf(base + ((double) base / 1000)).add(BigDecimal.valueOf(Long.MAX_VALUE));
+                BigDecimal b = IgniteTestUtils.randomBigDecimal(RND, scale, precision);
+                return b;
             case UUID:
-                return new UUID(base, base);
+                return new UUID(RND.nextLong(), RND.nextLong());
             case DURATION:
-                return Duration.ofNanos(base);
+                return Duration.ofNanos(RND.nextLong());
             case DATETIME:
                 return LocalDateTime.of(
-                        (LocalDate) generateValueByType(base, ColumnType.DATE),
-                        (LocalTime) generateValueByType(base, ColumnType.TIME)
+                        (LocalDate) generateValueByType(ColumnType.DATE, scale, precision),
+                        (LocalTime) generateValueByType(ColumnType.TIME, scale, precision)
                 );
             case TIMESTAMP:
                 return Instant.from(
-                        ZonedDateTime.of((LocalDateTime) generateValueByType(base, ColumnType.DATETIME), ZoneId.systemDefault()));
+                        ZonedDateTime.of((LocalDateTime) generateValueByType(ColumnType.DATETIME, scale, precision),
+                                ZoneId.systemDefault()));
             case DATE:
-                return LocalDate.of(2022, 01, 01).plusDays(base % 30);
+                return LocalDate.of(1900 + RND.nextInt(1000), 1 + RND.nextInt(12), 1 + RND.nextInt(28));
             case TIME:
-                return LocalTime.of(0, 00, 00).plusSeconds(base % 1000);
+                return IgniteTestUtils.randomTime(RND, scale);
```

Comment:
you fell into your own trap: scale != 0 at this place only in case of direct invocations of this method with magic constants as arguments.

Reply:
you are right. Thanks